astropy/io/fits/tests/test_table.py

```diff
@@ -3377,12 +3377,14 @@ def test_column_format_interpretation(self):
         """
 
         for recformat, fitsformat in NUMPY2FITS.items():
+            if recformat == "a":
+                fitsformat = "0A"
```

Comment:
Looking at this change, it's apparently problematic to edit ``NUMPY2FITS`` to change it to ``{"a":  "0A"}``.
Part of the problem appears to be that ``np.dtype("a") `` won't get the correct ``.format`` to compare to `fitsformat`. Am I understanding the problem correctly?